src/service/viewer-impl.js

```diff
@@ -426,6 +432,10 @@ export class Viewer {
       this.lastVisibleTime_ = now;
       this.hasBeenVisible_ = true;
       this.whenFirstVisibleResolve_();
+      if (this.nextVisibleResolve_) {
```

Comment:
How about doing this similar to `whenFirstVisible`? ie.

```
...
    this.whenNextVisiblePromise_ = new Promise(resolve => {
      this.whenNextVisibleResolve_ = resolve;
    });
....

...
// no need to check nullity of this. whenNextVisibleResolve_ anymore.
// just call and nullify nextVisiblePromise_
this.whenNextVisibleResolve_();
this.nextVisiblePromise_ = null;
...
```

Reply:
Hmmmm unlike `whenFirstVisible` `whenNextVisible` return a promise that should be reset everytime doc becomes invisible. 
I am trying to not initiate a promise if no one is requesting `whenNextVisible`. Otherwise it would require use to create a new promise on every visible state change. 

I did add a `whenNextVisibleResolve_()` helper function to make `#onVisiblityChange_` cleaner. I would prefer to not add that though... let me know